Sources/SymbolKit/SymbolGraph/Symbol/Symbol+KindIdentifier.swift

```diff
@@ -125,14 +99,19 @@ extension SymbolGraph.Symbol {
         /// - Parameter identifier: The identifier string to parse.
         public init(identifier: String) {
             // Check if the identifier matches a symbol kind directly.
-            if let firstParse = Self.lookupIdentifier(identifier: identifier) {
+            if let firstParse = KindIdentifier.lookupIdentifier(identifier: identifier) {
                 self = firstParse
             } else {
                 // For symbol graphs which include a language identifier with their symbol kinds
                 // (e.g. "swift.func" instead of just "func"), strip off the language prefix and
                 // try again.
                 let cleanIdentifier = KindIdentifier.cleanIdentifier(identifier)
-                self = Self.lookupIdentifier(identifier: cleanIdentifier) ?? .unknown
+                if let secondParse = KindIdentifier.lookupIdentifier(identifier: cleanIdentifier) {
+                    self = secondParse
+                } else {
+                    // If we still don't have a match, store the whole original string.
+                    self = .init(rawIdentifier: identifier)
```

Comment:
A small suggestion:
Line 112 use `self = .init(rawIdentifier: identifier)` in `public init(identifier: String)`
Line 120 use `self = KindIdentifier(identifier: identifier)` in ` public init(from decoder: Decoder)`

Can we use the same style: Either `self = KindIdentifer(someinitMethod)`, `self = .init(someinitMethod)` or `self.init(someinitMethod)`

And according the doc here [Initialization](https://docs.swift.org/swift-book/LanguageGuide/Initialization.html).

It seems `self.init(someinitMethod)` is the preferred one.

```suggestion
                    self.init(rawIdentifier: identifier)
```